app/main/helpers/__init__.py

```diff
@@ -8,3 +11,13 @@ def hash_email(email):
     m.update(email.encode('utf-8'))
 
     return base64.urlsafe_b64encode(m.digest())
+
+
+def supplier_role_required(func):
```

Comment:
I did it that way to start with but then decided to stick with the Flask `login_required`.  But yep, it's probably nicer to just have the one decorator.

My main worry was that it could be confusing once we also have `login_required` routes in the buyer app they will be calling different methods.